packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp

```diff
@@ -1518,6 +1524,8 @@ void mult_AT_B_newmatrix(
       bool isMM = slist.get("isMatrixMatrix_TransferAndFillComplete",false);
       int mm_optimization_core_count=3000; // ~3000 for serrano
```

Comment:
Please pull the default value from `::Tpetra::Details::Behavior::TAFC_OptimizationCoreCount()`.  This will ensure that (a) there is a single source for the default value, no matter the code path, and (b) the relevant environment variable, if provided, always overrides the compiled-in default value.